dht/sharder.hh

```diff
@@ -86,6 +142,9 @@ public:
             , _start_boundary(_sharder.shard_of(_start_token) == shard ?
                 _range.start() : range_bound<dht::token>(_sharder.token_for_next_shard(_start_token, shard))) {
     }
+    // Returns the next token_range that is both wholly contained within the input range and also
+    // wholly owned by the input shard_id. When the input range is exhausted, std::nullopt is returned.
+    // Note if the range does not intersect the shard at all, it may be returned immediately.
```

Comment:
Probably cleaner to say "std::nullopt may be returned immediately" instead of "it can be returned immediately"